### PR TITLE
Add CSP and other files to explorer

### PR DESCRIPTION
**Pull request by ty-d**
Add "CSP Files" and "Other" section to the ObjectScript explorer, and support compilation of those files. The studio actions editor context menu was modified to appear for CSP and other files.

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -24,6 +24,16 @@ export class AtelierAPI {
     return workspaceState.get(this.workspaceFolder + ":iris", false);
   }
 
+  private formatCspName(filename: string): string {
```
Comment (gjsjohnmurray) [inline src/api/index.ts:27]: @ty-d mayber rename this `transformNameIfCsp` for greater clarity.

**Comment by gjsjohnmurray** [comment]
> 
> 
> @ty-d how about fixing the linting warnings about missing return types in `index.ts` while you are touching it?

Looks like @daimor is already dealing with these in #172

File: src/explorer/models/rootNode.ts [hunk 2]
```diff
@@ -93,7 +122,15 @@ export class RootNode extends NodeBase {
             case "0":
             case "1":
             case "2":
+            case "3":
+            case "11":
               return new RoutineNode(el.Name, el.fullName, this.options);
+            case "100":
+              return new ClassNode(el.Name, el.fullName, this.options);
```
Comment (daimor) [inline src/explorer/models/rootNode.ts:129]:
```suggestion
```

File: src/explorer/models/rootNode.ts [hunk 3]
```diff
@@ -93,7 +122,15 @@ export class RootNode extends NodeBase {
...
+            case "10":
+              return new RootNode(el.Name, el.fullName, "dataNode:CSPApplication", this._category, this.options, true);
+            case "5":
+              return new ClassNode(el.Name, el.fullName, this.options);
```
Comment (daimor) [inline src/explorer/models/rootNode.ts:133]:
```suggestion
```

File: src/explorer/models/rootNode.ts [hunk 4]
```diff
@@ -84,6 +104,15 @@ export class RootNode extends NodeBase {
...
+            return el.Type === "100";
+          } else if (category === "CSP") {
+            return el.Type === "10" || el.Type === "5";
+          } else {
+            return true;
+          }
+        })
         .map((el) => {
           switch (el.Type) {
             case "9":
```
Comment (daimor) [inline src/explorer/models/rootNode.ts:118]:
```suggestion
            case "5":
            case "9":
            case "100":
```

File: src/explorer/models/rootNode.ts [hunk 5]
```diff
@@ -42,14 +51,20 @@ export class RootNode extends NodeBase {
...
       case "INC":
         spec = "*.inc";
         break;
       case "ALL":
         spec = "*.cls,*.mac,*.int,*.inc";
         break;
+      case "CSP":
+        spec = "*";
+        break;
+      case "OTHER":
```
Comment (daimor) [inline src/explorer/models/rootNode.ts:65]: OTHER replace everywhere with OTH, to be consistent with Atelier API

File: src/explorer/models/rootNode.ts [hunk 6]
```diff
@@ -42,14 +51,20 @@ export class RootNode extends NodeBase {
         spec = "*.cls";
         break;
       case "RTN":
-        spec = "*.mac,*.int";
+        spec = "*.mac,*.int,*.bas";
```
[the remark this answers is not in the document]
Comment (timleavitt) [reply src/explorer/models/rootNode.ts:54]: Why not?

**Comment by rajrsingh** [comment]
@daimor GitHub is saying you have requested changes, but I think your request has been addressed. Are we ready to merge this PR?

**Comment by daimor** [comment]
Nope, requested changes still have to be done.

**Pull request merged**